configure.ac

```diff
@@ -813,15 +821,22 @@ AS_IF([$natdynlink], [cmxs="cmxs"], [cmxs="cmx"])
 
 AC_DEFINE_UNQUOTED([OCAML_OS_TYPE], ["$ostype"])
 
-AS_IF([test -z "$PARTIALLD"],
+AC_CHECK_TOOL([DIRECT_LD],[ld])
```

Comment:
It's not strictly necessary to repeat this, given the `AC_CHECK_TOOLS` before `LT_INIT`